### PR TITLE
Improve tests on almalinux OSes

### PR DESCRIPTION
**Pull request by roussidis**
- Add necessary package `glibc-static` on almalinux 8 and almalinux 9
- Created a new function to get the kernel headers based on the linux distro.
- Moved the `Start sysbox` block after the initialization of the docker because it is dependend on the docker itself to find out the host filesystem
- Add an extra check regarding the host filesystem. If it is xfs idmapped_mounts is not supported.
- Add the correct kernel header paths in the tests.

### CODE REVIEW

File: tests/scr/testSysbox [hunk 1]
```diff
@@ -16,6 +16,7 @@ function idmapped_mount_supported() {
 	local rel_major=$(echo ${kernel_rel} | cut -d'.' -f1)
 	local rel_minor=$(echo ${kernel_rel} | cut -d'.' -f2)
 	[ ${rel_major} -gt 5 ] || ([ ${rel_major} -eq 5 ] && [ ${rel_minor} -ge 12 ])
+	[ $(docker info | grep "Backing Filesystem" | cut -d " " -f5) != "xfs" ] && return 0
```
Comment (ctalledo) [inline tests/scr/testSysbox:19]: Question: don't we need to link lines 18 & 19 into a single statement? What happens for example if line 18 returns false, would the function as a whole immediately return false (as it should) or would it continue into line 19 (which would be a mistake).

File: tests/scr/testSysboxCI [hunk 2]
```diff
@@ -16,6 +16,7 @@ function idmapped_mount_supported() {
 	local rel_major=$(echo ${kernel_rel} | cut -d'.' -f1)
 	local rel_minor=$(echo ${kernel_rel} | cut -d'.' -f2)
 	[ ${rel_major} -gt 5 ] || ( [ ${rel_major} -eq 5 ] && [ ${rel_minor} -ge 12 ] )
+	[ $(docker info | grep "Backing Filesystem" | cut -d " " -f5) != "xfs" ] && return 0
```
Comment (ctalledo) [inline tests/scr/testSysboxCI:19]: Same comment as in the testSysbox file.

**Review by ctalledo — APPROVED** [review]
Looks great in general, thanks @roussidis; just one comment / question.

File: tests/scr/testSysbox [hunk 3]
```diff
@@ -94,6 +93,8 @@ function run_all_tests() {
 
 	printf "\nSysbox health checking ...\n"
 	bats --tap tests/health/sysbox-health.bats
+	printf "\nExecuting life-cycle tests ... \n"
```
Comment (rodnymolina) [inline tests/scr/testSysbox:96]: Why is this change required? And if required, why moving these tests here change anything?

**Review by rodnymolina — COMMENTED** [review]
Question below.

File: tests/scr/testSysbox (hunk 1 above)
Comment (roussidis) [reply tests/scr/testSysbox:19]: I think you're right need to check this furthermore. But it might create some false positive scenarios. Since in line 19 we have a return statement we could swap 18 and 19 between them ? So if the filesystem is not xfs it will move on and check the kernel version. Maybe needs an improvement like 
```
[ $(docker info | grep "Backing Filesystem" | cut -d " " -f5) != "xfs" ] && return 0 || return 1
```
And then the lines could be 
```
	local rel_major=$(echo ${kernel_rel} | cut -d'.' -f1)
	local rel_minor=$(echo ${kernel_rel} | cut -d'.' -f2)
	[ $(docker info | grep "Backing Filesystem" | cut -d " " -f5) != "xfs" ] && return 0 || return 1
	[ ${rel_major} -gt 5 ] || ([ ${rel_major} -eq 5 ] && [ ${rel_minor} -ge 12 ])
```

I have not tested that but it could provide the exit factor that we need. So a testing scenario would be a system with the kernel lower than 5.12 and a non xfs filesystem, right ?

File: tests/scr/testSysbox (hunk 3 above)
Comment (roussidis) [reply tests/scr/testSysbox:96]: So sysbox while it was starting from the testContainerInit file was not respecting the filesystem type. That means that idmapped_mounts was trying to start up on an xfs filesystem which is not supported (yet). In that new [Start sysbox](https://github.com/nestybox/sysbox/pull/584/files/6b3a3328272465ea84e4064d30b1ad96d80ab5f5..98c8d7253ad3fdc7a779ae12652fe1e3888e0540#diff-ca142053720be2d2f7e2b6e884888c9637eb18a7e11a41a26449e4b36160ec91R292) function we do get the output of the host filesystem. If it is xfs for example sysbox is starting without idmapped_mounts `sysbox --disable-idmapped-mount`.

These changes were running as expected but the problem that we had to face with the `testSysbox` and the `run_all_tests` function is that the first tests restarted the sysbox which is reverting the previous checks. These tests were not "respecting" the flag that we set earlier and after the life-cycle tests were done it was like it was running `sysbox` with idmapped_mounts enabled once again. 

So while we discussed that with @ctalledo he proposed to move these tests last

File: tests/scr/testSysbox (hunk 1 above)
Comment (roussidis) [reply tests/scr/testSysbox:19]: Now that I am thinking it even more I don't think the aboce proposal would work. I would return solely the outcome of the filesystem type only.

File: tests/scr/testSysbox (hunk 3 above)
Comment (rodnymolina) [reply tests/scr/testSysbox:96]: @roussidis, thanks for the explanation, understood, I missed this conversation with @ctalledo while being on vacation.

Thanks again for your contribution!

Comment (roussidis) [reply tests/scr/testSysbox:96]: My pleasure to help !!
